src/data/set/intervals/pi.lean

```diff
@@ -106,12 +107,172 @@ end
 
 end pi_preorder
 
+section pi_partial_order
+variables [decidable_eq ι] [Π i, partial_order (α i)]
+
+lemma pi_image_update_Icc (f : Π i, α i) (i : ι) (a b : α i) :
```

Comment:
`set.pi_image_update_Icc` feels like a weird name for a lemma that does not mention `set.pi`.

Perhaps `set.image_function_update_Icc`?

Reply:
What about now? I originally had `pi.whatever`, then turned that into `set.pi_whatever` qince I noticed we could have the corresponding `finset` lemma. It now is `set.whatever` since I'm noticing the names need not mention `pi` as they already mention `update`/`single`.

Reply:
That's probably fine, as long as there's some precedent for referring to `single` and `update` without reference to `pi` and `function`.

Reply:
There sure is: [`finset.univ_sum_single`](https://leanprover-community.github.io/mathlib_docs/algebra/big_operators/pi.html#finset.univ_sum_single)

Reply:
Two more data points (not just the multiplicative version!) and I'm convinced :)

Reply:
`sum.elim_single_zero` and `update_le_iff`, say!

Reply:
I invoke `fin.cons_update` then.

Reply:
Works for me